intents/oversights/top_down_error.py

```diff
@@ -0,0 +1,247 @@
+        dimensions_len: Type-integer
+            size of the initial gfrouping dimensions.
+
+    Returns:
+        return a dictionary where each dictionary represent a debiasing 
+               suggestion according to the new column.
+    """
+    initial_num_rows = initial_result_table.shape[0]
+    new_num_rows = new_result_table.shape[0]
+
+    # changing dataframe to list of list to do next set of operation
+    initial_table_matrix = initial_result_table.values.tolist()
+    new_table_matrix = new_result_table.values.tolist()
+
+    initial_row_i = 0
+    new_row_i = 0
+
+    suggestion_row_list = []
+
+    # dominant percentage is the percentage of pairs where 
+    # (first value - second value) is positive.
+
+    while initial_row_i < initial_num_rows:
+        if initial_row_i == initial_num_rows - 1 or initial_table_matrix[initial_row_i][:dimensions_len] != initial_table_matrix[initial_row_i + 1][:dimensions_len]:
+            while new_row_i < new_num_rows and initial_table_matrix[initial_row_i][:dimensions_len] == new_table_matrix[new_row_i][:dimensions_len]:
+                new_row_i = new_row_i + 1
+        else:
+            new_max_corelation = 0
+            while new_row_i < new_num_rows and initial_table_matrix[initial_row_i][:dimensions_len] == new_table_matrix[new_row_i][:dimensions_len]:
+                if new_row_i == new_row_i - 1 or new_table_matrix[new_row_i][:dimensions_len + 1] != new_table_matrix[new_row_i + 1][:dimensions_len + 1]:
+                    max_disimilarity = constants.TOP_DOWN_ERROR_DISSIMILARITY_THRESHOLD
+                else:
+                    corelation = _calculate_relation(new_table_matrix[new_row_i][dimensions_len + 2], new_table_matrix[new_row_i + 1][dimensions_len + 2])
+                    new_max_corelation = max(corelation, new_max_corelation)
+                    new_row_i = new_row_i + 1
+                new_row_i = new_row_i + 1
+
+            initial_corelation = _calculate_relation(initial_table_matrix[initial_row_i][dimensions_len + 1], initial_table_matrix[initial_row_i + 1][dimensions_len + 1])
+
+            if new_max_corelation >= constants.TOP_DOWN_ERROR_DISSIMILARITY_THRESHOLD and initial_corelation <= constants.TOP_DOWN_ERROR_SIMILARITY_THRESHOLD:
+                suggestion_row_list.append(initial_row_i)
+                suggestion_row_list.append(initial_row_i + 1)
+            initial_row_i = initial_row_i + 1
+
+        initial_row_i = initial_row_i + 1
+
+    if len(suggestion_row_list) == 0:
+        return None
+    else:
+        new_suggestion = {}
+        new_suggestion['suggestion'] = 'both the value is almost same in some grouping but different if we add ' + new_added_column + ' in initial grouping.'
+        new_suggestion['oversight_name'] = 'top-down error'
+        new_suggestion['is_row_level_suggestion'] = True
+        new_suggestion['row_list'] = suggestion_row_list
+        return new_suggestion
```

Comment:
add key 'confidence_score' if feasible

Reply:
I haven't think about it. so I won't be able to add it right now.

Reply:
I have added confidence score as 100 right now.
It might be changed.